src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -290,73 +300,38 @@ private fun getMoreCompleteType(t1: Ty, t2: Ty): Ty {
 }
 
 private val RsBlock.type: Ty get() = expr?.type ?: TyUnit
+private val RsStructLiteralField.type: Ty? get() = declaration?.typeReference?.type
```

Comment:
`Ty?` seems smelly. We have a null-object, `TyUnknown`, and should probably use it here.